components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/DAO/GroupDAO.java

```diff
@@ -317,4 +319,37 @@ public void updateRoleName(int tenantId, String oldRoleName, String newRoleName)
         }
     }
 
+    public String[] getGroupNameList(String attributeName, String searchAttribute, Integer tenantId)
+            throws IdentitySCIMException {
+
+        List<String> roleList = new ArrayList<>();
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection();
+        PreparedStatement prepStmt = null;
+        ResultSet rSet = null;
+
+        try {
+            prepStmt = connection.prepareStatement(SQLQueries.LIST_SCIM_GROUPS_SQL_BY_ATT_AND_ATT_VALUE);
+            prepStmt.setInt(1, tenantId);
+            prepStmt.setString(2, attributeName);
+            prepStmt.setString(3, searchAttribute);
+
+            rSet = prepStmt.executeQuery();
+            while (rSet.next()) {
+                if (StringUtils.isNotEmpty(rSet.getString(1))) {
+                    roleList.add(rSet.getString(1));
+                }
+            }
+
+        } catch (SQLException e) {
+            log.error("Error when executing the SQL : " + SQLQueries.LIST_SCIM_GROUPS_SQL_BY_ATT_AND_ATT_VALUE);
```

Comment:
If you catch it and log it and rethrow it, then there's no way for the upstream code to know that you've already logged the exception, and so the same exception might get logged twice.

Reply:
Here, SQLException is converted into IdentitySCIMException. So inside the catch a log is printed because IdentitySCIMException is more generic. Is it unnecessary?